lib/install/inflate-shrinkwrap.js

```diff
@@ -26,7 +26,8 @@ module.exports = function (tree, swdeps, opts, finishInflating) {
   if (!npm.config.get('shrinkwrap') || !npm.config.get('package-lock')) {
     return finishInflating()
   }
-  tree.loaded = true
+  tree.loaded = false
+  tree.hasRequiresFromLock = Object.keys(swdeps).every((d) => swdeps[d].requires)
```

Comment:
well... no, if any of them are missing then we can't assume we know anything about the relationships between the modules.